### PR TITLE
Add copy constructor and test of sum aggregates

### PR DESCRIPTION
**Pull request by frreiss**
This PR adds a copy constructor to the `TensorArray` class. I added tests of the copy constructor and also threw in some tests of `sum` aggregations.

### CODE REVIEW

**Comment by frreiss** [comment]
Test failures on this PR are due to #47

File: text_extensions_for_pandas/array/test_tensor.py [hunk 1]
```diff
@@ -55,6 +55,10 @@ def test_create(self):
         with self.assertRaises(TypeError):
             TensorArray(2112)
 
+        # Copy constructor
+        s_copy = TensorArray(s)
```
Comment (BryanCutler) [inline text_extensions_for_pandas/array/test_tensor.py:59]: Is there a need to use like a copy constructor? I thinks it's more common to explicitly copy like `TensorArray(s).copy()`

Comment (frreiss) [reply text_extensions_for_pandas/array/test_tensor.py:59]: Makes sense. Updating this PR accordingly.

**Comment by frreiss** [comment]
Thanks for the review, @BryanCutler ! Pushed some changes to address review comments.

**Pull request merged**